### PR TITLE
chore(ast-spec): exclude tsx fixtures from type-check

### LINKED ISSUE #7166: Repo: v6 branch regression for adding to JSX.IntrinsicAttributes in typecheck task
### Suggestion

On the `v6` branch, we now get typecheck (`tsc`) complaints in `ast-spec`'s fixture for augmenting `JSX.IntrinsicElements`:

```plaintext
 ~/repos/typescript-eslint/packages/ast-spec $ yarn typecheck
yarn run v1.22.19
$ tsc -p tsconfig.json --noEmit
src/jsx/JSXNamespacedName/fixtures/component-dashed/fixture.tsx:8:24 - error TS2339: Property 'foo' does not exist on type 'JSX.IntrinsicElements'.

8 const componentBasic = <foo />;
                         ~~~~~~~

src/jsx/JSXNamespacedName/fixtures/component-dashed/fixture.tsx:9:25 - error TS2339: Property 'foo-bar:baz-bam' does not exist on type 'JSX.IntrinsicElements'.

9 const componentDashed = <foo-bar:baz-bam />;
                          ~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: src/jsx/JSXNamespacedName/fixtures/component-dashed/fixture.tsx:8
```

It happens both for traditional names (e.g. `foo`) and for TypeScript 5.1's new namespaced names (e.g. `'foo-bar:baz-bam'`) added in #7088. So I don't think it's specific to TS 5.1's namespace changes.

But... why is this happening? Why is the fixture not allowing augmentation of `JSX.IntrinsicElements`? 

I'm 80% confident this is a silly little mistake I made in merging `main` -> `v6`, but can't figure it out... :sob

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

- [x] Addresses an existing open issue: fixes #7166
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

- exclude tsx files from type-check
- allow tests to be executed on windows

### CODE REVIEW

File: packages/ast-spec/tests/fixtures.test.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ const FIXTURES: readonly Fixture[] = [...VALID_FIXTURES, ...ERROR_FIXTURES].map(
         }
       })(),
       ext,
-      isError: absolute.includes('/_error_/'),
+      isError: /[\\/]_error_[\\/]/.test(absolute),
```
Comment (bradzacher) [inline packages/ast-spec/tests/fixtures.test.ts:75]: what's the difference between these two? Seem like they're the same?

Comment (armano2) [reply packages/ast-spec/tests/fixtures.test.ts:75]: No, windows has backslash and Linux/Mac/Unix has slash

Comment (armano2) [reply packages/ast-spec/tests/fixtures.test.ts:75]: @bradzacher is there anything blocking this change?

Comment (bradzacher) [reply packages/ast-spec/tests/fixtures.test.ts:75]: ah - i wasn't paying attention - it's now a regex literal so the `\\` means `\` - it's not escaping the backslash to escape the `/`.

**Review by bradzacher — APPROVED** [review]

**Pull request merged**